frontend/src/modules/redirect/RedirectPage.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { FC } from 'react';
+import { parse } from 'query-string';
+import { useDispatch, useSelector } from 'react-redux';
+import { Redirect } from 'react-router-dom';
+
+import { loginWithGithub } from 'xbddReducer';
+import Loading from 'modules/loading/Loading';
+import { RootStore } from 'rootReducer';
+
+const getCodeFromQueryParam = (): string | undefined => {
+  const { location } = window;
+  const search = location.search.slice(1);
+  const { code } = parse(search);
+  const extractedCode = Array.isArray(code) ? code[0] : code;
+  return extractedCode || undefined;
+};
+
+const RedirectPage: FC = () => {
+  const user = useSelector((store: RootStore) => store.app.user);
+  const dispatch = useDispatch();
+  const code = getCodeFromQueryParam();
+
+  if (code && !user) {
+    dispatch(loginWithGithub(code));
+  }
+
+  if (user) {
+    return <Redirect to="/" />;
+  }
+
+  return <Loading loading={true} />;
```

Comment:
It’s the page that github redirects to one the login has been processed for their site. Basically what is happening is after Github redirects here there is a code in the URL. That is pulled out and login (to xbdd) is dispatched so we render a loading spinner while this is happening. Once that is processed there will be a user in redux so we redirect to the home screen and the summary page will then be rendered. 

I don’t see a way to actually split this further but it’s possible we could rename it. Redirect makes sense to me but I see it as the OAuth redirect workflow. Ideally this would handle any OAuth redirect we had but now that I think about it maybe we would need one for each provider. Would GithubRedirect be better or still too vague?

Reply:
Yep ok, if it's for logins then maybe LoginRedirect.tsx? Or GithubLoginRedirect if will be provider-specific?